chain/x/zoracle/handler.go

```diff
@@ -155,6 +155,11 @@ func handleMsgRequestData(ctx sdk.Context, keeper Keeper, msg MsgRequestData) sd
 		return err.Result()
 	}
 
+	err = keeper.ConsumeGasForExecute(ctx, id, msg.ExecuteGas)
```

Comment:
IMO the consume gas can just happen in `AddRequest`. By doing that also avoid unnecessary load-and-update of `Request`.